Dan update dapp (#954)
substrate-developer-hub/substrate-developer-hub.github.io: docs/tutorials/build-a-dapp/pallet.md

+-- ...
```

You will see some pre-written code that acts as a template for a new pallet.
When writing your own pallets in the future, you will likely find the scaffolding
in this template pallet useful. But for the purposes of really learning how a pallet
is constructed, **delete all contents of this file**.

> > The following sections will start _from scratch_ - assuming that you have cleared the

Reviewer: This phrase seems repetitive to me: "We will start from scratch for full showing how to build a pallet from ground zero." I like the note here though, maybe try squash the two phrases together?

Author: open to your language 😁

+-- ...
```

You will see some pre-written code that acts as a template for a new pallet.
When writing your own pallets in the future, you will likely find the scaffolding
in this template pallet useful. But for the purposes of really learning how a pallet
is constructed, **delete all contents of this file**.

> The following sections will start _from scratch_ - assuming that you have cleared the

> > existing `pallet/template/src/lib.rs` file. We will start from scratch for full showing

Reviewer: ```suggestion
> existing `pallet/template/src/lib.rs` file. 
```

> > how to build a pallet from ground zero.

Reviewer: ```suggestion
```

blockchain development. We will show you what each of these components looks like for a basic Proof
Of Existence pallet.

> In the following sections, we have the required code blocks you can **copy and paste** into your

Reviewer: Reads a little convoluted to me.
**Rephrase suggestion:** In the following sections, the code blocks required for the PoE pallet are provided for you to simply **copy and paste** into the empty `pallet/template/src/lib.rs` file.

Author: go ahead and change as you see fit!

blockchain development. We will show you what each of these components looks like for a basic Proof
Of Existence pallet.

> In the following sections, we have the required code blocks you can **copy and paste** into your

Reviewer: ```suggestion
In the following sections, the code blocks required for the PoE pallet are provided for you to simply copy and paste into the empty pallet/template/src/lib.rs file.
```

> now empty `pallet/template/src/lib.rs` file _in order_ to create this PoE pallet.

Reviewer: ```suggestion

```

use sp_std::vec::Vec;
```

> > Critical: the `no_std` feature is **required** for all pallets! This is because we are building a

Reviewer: For consistency in style, I'd suggest to bold "Critical" :)

Author: Sure - messaged you in element. Feel free to make edits

> _runtime_ module that _must_ compile to WASM, and therefore cannot depend on rust's `std` 
> dependencies.

Most of these imports are already available because they were used in the template pallet whose code
we just deleted. However, `sp_std` is not available and we need to list it as a dependency.

**Add** these lines to your `pallets/template/Cargo.toml` file under `[dependencies]` and 
`[features]` sections, as shown below:

**`pallets/template/Cargo.toml`**

```toml
[dependencies]

#--snip--

sp-std = { default-features = false, version = '3.0.0' }  <-- Add this line

#--snip--

[features]
default = ['std']
std = [
    #--snip--
    'sp-std/std',                                         <-- Add this line
]
```

> Notice here that all _runtime specific_ dependencies _must_ have default feature (meaning `std`)
> disabled so we can compile the runtime to WASM.You can also learn more about why this is 
> necessary in the [Add a Pallet](../add-a-pallet) tutorial.
>

> > You _can_ use `std` features in non-runtime components like `mock.rs` and `tests.rs` using 

Reviewer: I like the references here. Nice. Just noticed a missing space between sentences :)

Reviewer: Also - this could be something to reference additionally: https://substrate.dev/docs/en/knowledgebase/advanced/executor#wasm-execution

Author: fix it?

use sp_std::vec::Vec;
```

> > Critical: the `no_std` feature is **required** for all pallets! This is because we are building a

Reviewer: ```suggestion
> **Critical**: the `no_std` feature is **required** for all pallets! This is because we are building a
```

> _runtime_ module that _must_ compile to WASM, and therefore cannot depend on rust's `std` 
> dependencies.

Most of these imports are already available because they were used in the template pallet whose code
we just deleted. However, `sp_std` is not available and we need to list it as a dependency.

**Add** these lines to your `pallets/template/Cargo.toml` file under `[dependencies]` and 
`[features]` sections, as shown below:

**`pallets/template/Cargo.toml`**

```toml
[dependencies]

#--snip--

sp-std = { default-features = false, version = '3.0.0' }  <-- Add this line

#--snip--

[features]
default = ['std']
std = [
    #--snip--
    'sp-std/std',                                         <-- Add this line
]
```

> Notice here that all _runtime specific_ dependencies _must_ have default feature (meaning `std`)

> > disabled so we can compile the runtime to WASM.You can also learn more about why this is 

Reviewer: ```suggestion
> disabled so we can compile the runtime to WASM. You can also learn more about why this is 
```

removed. Note that convention is to include an array with descriptive names for these parameters at
the end of event documentation.

> ## Pallet Errors

Reviewer: This should probably be a ### heading

removed. Note that convention is to include an array with descriptive names for these parameters at
the end of event documentation.

> ## Pallet Errors

Reviewer: ```suggestion
### Pallet Errors
```
